components/layout/wrapper.rs

```diff
@@ -681,42 +691,36 @@ pub trait ThreadSafeLayoutNode : Clone + Copy + Sized + PartialEq {
     fn as_element(&self) -> Self::ConcreteThreadSafeLayoutElement;
 
     #[inline]
-    fn get_pseudo_element_type(&self) -> PseudoElementType<display::T>;
+    fn get_pseudo_element_type(&self) -> PseudoElementType<Option<display::T>>;
 
     #[inline]
     fn get_before_pseudo(&self) -> Option<Self> {
-        self.borrow_layout_data().unwrap()
-            .style_data.per_pseudo
-            .get(&PseudoElement::Before)
-            .map(|style| {
-                self.with_pseudo(PseudoElementType::Before(style.get_box().display))
-            })
+        if self.borrow_layout_data().unwrap()
+               .style_data.per_pseudo
+               .contains_key(&PseudoElement::Before) {
+            Some(self.with_pseudo(PseudoElementType::Before(None)))
+        } else {
+            None
+        }
     }
 
     #[inline]
     fn get_after_pseudo(&self) -> Option<Self> {
-        self.borrow_layout_data().unwrap()
-            .style_data.per_pseudo
-            .get(&PseudoElement::After)
-            .map(|style| {
-                self.with_pseudo(PseudoElementType::After(style.get_box().display))
-            })
+        if self.borrow_layout_data().unwrap()
+               .style_data.per_pseudo
+               .contains_key(&PseudoElement::Before) {
```

Comment:
Whoops, what a embarrasing typo, `PseudoElement::After` should be here, there should be no more test failures after that I think.